torch/_inductor/fx_passes/split_cat.py

```diff
@@ -586,6 +593,89 @@ def erase_old_nodes(self, graph, split_node, next_users):
             graph.erase_node(node)
 
 
+class GetItem(CallFunction):
+    def __init__(self, arg, index):
+        super().__init__(operator.getitem, arg, index)
+
+    def find_anchor_nodes(self, ctx: MatchContext, searched, matching_nodes=True):
+        # We generally match GetItem with arg being an Arg(). So, we never return the anchor
+        # nodes as the stored node in ctx.pattern_to_node is returned. Here we override find_anchor_nodes
+        # to not use ctx.pattern_to_node
+        for pattern in self.flat_args_kwargs[0]:
+            if isinstance(pattern, PatternExpr):
+                for other_node in pattern.find_anchor_nodes(ctx, searched):
+                    if not isinstance(other_node, torch.fx.Node):
+                        continue
+                    for node in other_node.users:
+                        if node not in searched:
+                            if not matching_nodes or self._match_fns(node):
```

Comment:
Can you explain what matching_nodes does?

Reply:
If you have a better way of doing this please suggest. I was initially thinking of using a different method like `find_sibling_nodes`, but it seemed very similar to `find_anchor_nodes` in implementation.

Reply:
btw, I just realized I forgot to add this variable to some other classes. I can add them, but waiting for your response first.

Reply:
This isn't really what `find_anchor_nodes()` is meant for, I worry this confuses the semantics.

I'd rather see an extra check, either in the `_match` function or in the handler that is just based on the users of split.

Reply:
Sure, I can do that